### PR TITLE
Editors googlespreadsheet csv to bigquery

### PR DESCRIPTION
**Pull request by tayowonibi**
New pipeline created, which import google spreadsheet to bigquery - which is basically the first step to address the issue https://github.com/elifesciences/issues/issues/4864.  @de-code :Can you check the schema to confirm that it basically addresses the data requirement.

Associated pipeline in nifi is called editors-google-spreadsheet-deposit --staging

### CODE REVIEW

File: flows/ejp-csv-deposit/schemas/1119_datascience_manuscript_reviews.json [hunk 1]
```diff
@@ -0,0 +1,60 @@
+[
```
Comment (de-code) [inline flows/ejp-csv-deposit/schemas/1119_datascience_manuscript_reviews.json]: probably doesn't belong into this PR

Comment (tayowonibi) [reply flows/ejp-csv-deposit/schemas/1119_datascience_manuscript_reviews.json]: git commits from the reviewer comments import somewhat got transferred to the spreadsheet import branch when i switched the project, should i try clean this up?

**Comment by de-code** [comment]
I added a question for Maria on the issue. The `Senior Editors` tab has two alternatives tabs (one is blank and not currently include in the schema). We should be able to change the Google Sheet. We will also need some indicator from which tab the data is coming from. i.e. whether it's a `BRE`, `Senior Editor` or `Guest Editor`.

File: flows/ejp-csv-deposit/schemas/1119_datascience_manuscript_reviews.json (hunk 1 above)
Comment (de-code) [reply flows/ejp-csv-deposit/schemas/1119_datascience_manuscript_reviews.json]: Would be good to have the PR include only the related files and keep the other CSV in a separate PR. You could just remove the file and the related changes?

I always use `Squash Merge`.

**Comment by tayowonibi** [comment]
> I added a question for Maria on the issue. The `Senior Editors` tab has two alternatives tabs (one is blank and not currently include in the schema). We should be able to change the Google Sheet. We will also need some indicator from which tab the data is coming from. i.e. whether it's a `BRE`, `Senior Editor` or `Guest Editor`.

Indicator for the source sheet of the editors file (i.e. BRE, Senior Editor, Guest Editor) is stored in 'status' column of the schema

**Comment by de-code** [comment]
> > I added a question for Maria on the issue. The `Senior Editors` tab has two alternatives tabs (one is blank and not currently include in the schema). We should be able to change the Google Sheet. We will also need some indicator from which tab the data is coming from. i.e. whether it's a `BRE`, `Senior Editor` or `Guest Editor`.
> 
> Indicator for the source sheet of the editors file (i.e. BRE, Senior Editor, Guest Editor) is stored in 'status' column of the schema

Okay, fair enough. Maybe another name would be good?

What table is that imported to, can I look at it in BigQuery from your dev pipeline?

**Comment by tayowonibi** [comment]
public_editors_profile in staging

**Comment by de-code** [comment]
The source spreadsheet has been updated (columns / sheets removed). It sounds like we not expecting the spreadsheet to change any time soon.

**Comment by lsh-0** [comment]
please cc me in any PRs you open against this repo. I don't need to be added as a reviewer but I'd like to keep a close eye on any changes for the time being.

**Comment by tayowonibi** [comment]
@de-code - I made some modifications to the branch for this PR. Can you help me review it and test  before i merge it into the master

**Comment by tayowonibi** [comment]
> @de-code - I made some modifications to the branch for this PR. Can you help me review it and test before i merge it into the master

@de-code  - just wondering if you have a chance to review this code. 
/cc  @lsh-0  - just adding you now because i didn't cc you in the original PR

File: flows/ejp-csv-deposit/schemas/1119_datascience_manuscript_reviews.json (hunk 1 above)
Comment (lsh-0) [reply flows/ejp-csv-deposit/schemas/1119_datascience_manuscript_reviews.json]: > I always use Squash Merge.

I prefer not losing git history with a squash. I don't mind if commits are messy or trivial or spurious. There is no policy about this at elife, it comes down to personal preference.

> Would be good to have the PR include only the related files and keep the other CSV in a separate PR.

:+1: regardless of to-squash-or-not, keep PRs focused on one feature or one fix at a time and aim for many small incremental improvements over time.

Comment (de-code) [reply flows/ejp-csv-deposit/schemas/1119_datascience_manuscript_reviews.json]: > > I always use Squash Merge.
> 
> I prefer not losing git history with a squash. I don't mind if commits are messy or trivial or spurious. There is no policy about this at elife, it comes down to personal preference.

We actually had discussions regarding that on [Slack #tech-team](https://elifesciences.slack.com/archives/C498YPDGT/p1541433729006100) and I understood there was an agreement to generally squash.

Comment (tayowonibi) [reply flows/ejp-csv-deposit/schemas/1119_datascience_manuscript_reviews.json]: so, what exactly needs to be done to move forward with approval of this PR (i am a bit lost on the specifics wrt the above discussion)

**Comment by de-code** [comment]
> > Indicator for the source sheet of the editors file (i.e. BRE, Senior Editor, Guest Editor) is stored in 'status' column of the schema
> 
> Okay, fair enough. Maybe another name would be good?

I am still thinking another name could maybe be more descriptive. e.g. `editor_type`, `editor_role` (it is kind of matches the role we use elsewhere, only that `BRE` should be `Reviewing Editor`).
 
>> What table is that imported to, can I look at it in BigQuery from your dev pipeline?
>
> public_editors_profile in staging

Would you mind updating the table with the latest changes please? (including the schema)

**Comment by tayowonibi** [comment]
> > > Indicator for the source sheet of the editors file (i.e. BRE, Senior Editor, Guest Editor) is stored in 'status' column of the schema
> > 
> > 
> > Okay, fair enough. Maybe another name would be good?
> 
> I am still thinking another name could maybe be more descriptive. e.g. `editor_type`, `editor_role` (it is kind of matches the role we use elsewhere, only that `BRE` should be `Reviewing Editor`).
> 
> > > What table is that imported to, can I look at it in BigQuery from your dev pipeline?
> > 
> > 
> > public_editors_profile in staging
> 
> Would you mind updating the table with the latest changes please? (including the schema)

Done

File: flows/ejp-csv-deposit/update-schemas.sh [hunk 2]
```diff
@@ -21,3 +21,4 @@ dump 767_datascience_person_roles
 dump 779_datascience_reviewer_info_all
 # reports 795 and 796 both go to the same table
 dump 795_datascience_person_merge_info_all
+dump 1119_datascience_manuscript_reviews
```
Comment (de-code) [inline flows/ejp-csv-deposit/update-schemas.sh]: Should this line also be removed from the PR?

**Comment by de-code** [comment]
> > Would you mind updating the table with the latest changes please? (including the schema)
> 
> Done

With data please? :-) (It seems to be empty)

**Comment by tayowonibi** [comment]
> With data please? :-) (It seems to be empty)

this pipeline/script will load the table with data.

**Comment by de-code** [comment]
> > With data please? :-) (It seems to be empty)
> 
> this pipeline/script will load the table with data.

Okay, I thought you were usually running it locally.

File: flows/ejp-csv-deposit/update-schemas.sh [hunk 3]
```diff
@@ -20,4 +20,4 @@ dump 767_datascience_person_roles
 # reports 779 and 780 both go to the same table
 dump 779_datascience_reviewer_info_all
 # reports 795 and 796 both go to the same table
-dump 795_datascience_person_merge_info_all
+dump 795_datascience_person_merge_info_all
```
Comment (de-code) [inline flows/ejp-csv-deposit/update-schemas.sh]: minor: files usually end with a line feed (which is why this file still shows up as a change - and the indicator for no new line in GitHub)

File: flows/ejp-csv-deposit/update-schemas.sh [hunk 4]
```diff
@@ -5,7 +5,7 @@ function dump {
     project="elife-data-pipeline"
     dataset="nifidemo4_temp"
     table=$1
-    echo "$table"
+    echo "$table "
```
Comment (de-code) [inline flows/ejp-csv-deposit/update-schemas.sh]: minor: I guess you added the space by accident?

**Comment by tayowonibi** [comment]
> > > With data please? :-) (It seems to be empty)
> > 
> > 
> > this pipeline/script will load the table with data.
> 
> Okay, I thought you were usually running it locally.

data is in staging now

**Comment by tayowonibi** [comment]
@de-code - to squash and merge?

**Review by de-code — APPROVED** [review]
looks good

**Pull request merged**